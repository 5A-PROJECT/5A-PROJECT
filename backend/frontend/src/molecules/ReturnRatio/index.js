import React from 'react';
import PropTypes from 'prop-types';
import ArrowDropUpIcon from '@material-ui/icons/ArrowDropUp';
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
import ArrowRightIcon from '@material-ui/icons/ArrowRight';
import styled from 'styled-components';
import { StylesProvider } from '@material-ui/core';

const UpIcon = styled(ArrowDropUpIcon)`
  color: red;
  font-size: ${(props) => props.size};
`;

const DownIcon = styled(ArrowDropDownIcon)`
  color: blue;
  font-size: ${(props) => props.size};
`;

const StableIcon = styled(ArrowRightIcon)`
  color: grey;
  font-size: ${(props) => props.size};
`;

const ReturnRatioWrapper = styled.div`
  display: flex;
  align-items: center;
  font-weight: bold;
  min-width: 75px;
`;

const Ratio = styled.span`
  font-size: ${(props) => props.fontSize};
  margin-left: 0.3rem;
`;

function ReturnRatio({ ratio, fontSize = '1.5rem', iconSize = '1.5rem' }) {
  return (
    <ReturnRatioWrapper>
      <StylesProvider injectFirst>
        {ratio > 0 && <UpIcon size={iconSize} title="up" />}
        {ratio < 0 && <DownIcon size={iconSize} title="down" />}
        {parseFloat(ratio) === 0 && (
          <StableIcon size={iconSize} title="stable" />
        )}
      </StylesProvider>
      <Ratio fontSize={fontSize}>{ratio}%</Ratio>
    </ReturnRatioWrapper>
  );
}

ReturnRatio.propTypes = {
  ratio: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  fontSize: PropTypes.string,
  iconSize: PropTypes.string,
};

export default ReturnRatio;
