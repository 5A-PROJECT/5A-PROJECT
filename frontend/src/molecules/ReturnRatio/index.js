import React from 'react';
import PropTypes from 'prop-types';
import ArrowDropUpIcon from '@material-ui/icons/ArrowDropUp';
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
// import TrendingDownIcon from '@material-ui/icons/TrendingDown';
// import TrendingUpIcon from '@material-ui/icons/TrendingUp';
import styled from 'styled-components';
import { StylesProvider } from '@material-ui/core';

const UpIcon = styled(ArrowDropUpIcon)`
  color: red;
  font-size: ${(props) => props.size};
`;

const DownIcon = styled(ArrowDropDownIcon)`
  color: blue;
  font-size: ${(props) => props.size};
`;

const ReturnRatioWrapper = styled.div`
  display: flex;
  align-items: center;
  font-weight: bold;
`;

const Ratio = styled.span`
  font-size: ${(props) => props.fontSize};
  margin-left: 0.3rem;
`;

function ReturnRatio({ ratio, fontSize = '1.5rem', iconSize = '1.5rem' }) {
  /* // -뗄려면 이거 쓰셈
    const getPositive = (profit) => {
    if (profit < 0) {
      return -profit;
    }
    return profit;
  };

  const positiveRatio = useMemo(() => getPositive(ratio), [ratio]);  
  */

  return (
    <ReturnRatioWrapper>
      <StylesProvider injectFirst>
        {ratio > 0 ? <UpIcon size={iconSize} /> : <DownIcon size={iconSize} />}
      </StylesProvider>
      <Ratio fontSize={fontSize}>{ratio}%</Ratio>
    </ReturnRatioWrapper>
  );
}

ReturnRatio.propTypes = {
  ratio: PropTypes.string.isRequired,
  fontSize: PropTypes.string,
  iconSize: PropTypes.string,
};

export default ReturnRatio;
