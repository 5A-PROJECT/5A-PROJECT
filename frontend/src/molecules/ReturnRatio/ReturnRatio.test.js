import React from 'react';
import { render } from '@testing-library/react';
import ReturnRatio from '.';

const Sample = (ratio) => render(<ReturnRatio ratio={ratio} />);

describe('<ReturnRatio />', () => {
  it('스냅샷과 일치해야 합니다.', () => {
    const { container } = Sample('3.5');
    expect(container).toMatchSnapshot();
  });
  it('음수 수익률은 DownIcon과 -{ratio}% 가 렌더링 되어야 합니다.', () => {
    const { getByText, getByTitle } = Sample('-3.5');
    getByTitle('down');
    getByText('-3.5%');
  });
  it('양수 수익률은 UpIcon 과 {ratio}% 가 렌더링 되어야 합니다.', () => {
    const { getByText, getByTitle } = Sample('55.5');
    getByTitle('up');
    getByText('55.5%');
  });
});
