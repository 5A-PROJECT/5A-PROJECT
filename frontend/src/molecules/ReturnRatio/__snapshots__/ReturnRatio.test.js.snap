// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReturnRatio /> 스냅샷과 일치해야 합니다. 1`] = `
<div>
  <div
    class="sc-AxhCb cvKNLU"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root sc-AxjAm eaAYzN"
      focusable="false"
      title="up"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 14l5-5 5 5z"
      />
    </svg>
    <span
      class="sc-AxhUy engnwa"
      font-size="1.5rem"
    >
      3.5
      %
    </span>
  </div>
</div>
`;
